app/javascript/components/physical-storage-form/index.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import MiqFormRenderer from '../../forms/data-driven-form';
+import createSchema from './physical-storage-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+
+const PhysicalStorageForm = ({ redirect }) => {
+  const state = useState(undefined);
+
+  const onSubmit = async(values) => {
+    miqSparkleOn();
+    const message = __('Physical storage added');
+    API.post('/api/physical_storages', { action: 'create', resource: values })
+      .then(() => miqRedirectBack(message, 'success', redirect)).catch((_) => {
+        miqSparkleOff();
+      });
+  };
+
+  const onCancel = () => {
+    const message = __('Adding of physical storage was cancelled by the user');
+    miqRedirectBack(message, 'success', redirect);
+  };
+
+  return (
+    <MiqFormRenderer
+      schema={createSchema(...state)}
+      onSubmit={onSubmit}
+      onCancel={onCancel}
+      buttonsLabels={{
+        submitLabel: window.__('Add'),
+        cancelLabel: window.__('Cancel'),
+      }}
+    />
+  );
+};
+
+
+PhysicalStorageForm.propTypes = {
+  redirect: PropTypes.string,
```

Comment:
I guess this attribute can be set to required:
```
redirect: PropTypes.string.isRequired,
```
Then you can delete the `defaultProps` from below completely.